minetest/minetest
# Settings GUI: Hide advanced settings by default

Part of  #13476

Adds a new checkbox to show/hide advanced settings, hidden by default. By hiding advanced settings, the settings GUI is less overwhelming.

![image](https://github.com/minetest/minetest/assets/2122943/434fcdc2-ee39-483f-9f4c-5eff6e0f7fde)

## To do

This PR is Ready for Review.

Questions:

- Resolve the ambiguity between different types of advanced settings (there's both an "Advanced" section and the advanced checkbox now)
- Finalise which settings are considered advanced

## How to test

* Run MT, open settings
* Toggle show advanced settings dialog, look at settings
* See all settings marked as advanced in settingtypes.txt
